src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.cpp

```diff
@@ -147,20 +143,17 @@ std::set<std::vector<element::Type>> jit_brgemm_emitter::get_supported_precision
     }
 }
 
-void jit_brgemm_emitter::init_brgemm_kernel(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>& kernel, bool use_amx) {
+void jit_brgemm_emitter::init_brgemm_kernel(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>& kernel) {
     brgemm_t desc;
     const bool is_int8 = utils::one_of(ctx.dt_in0, data_type::u8, data_type::s8) && utils::one_of(ctx.dt_in1, data_type::u8, data_type::s8);
-    auto isa = use_amx ? isa_undef
-                       : ctx.dt_in0 == dnnl_data_type_t::dnnl_bf16 ? avx512_core_bf16 : (is_int8 ? avx512_core_vnni : avx512_core);
+    auto isa = ctx.is_with_amx ? isa_undef
+                               : ctx.dt_in0 == dnnl_data_type_t::dnnl_bf16 ? avx512_core_bf16 : (is_int8 ? avx512_core_vnni : avx512_core);
```

Comment:
I plunged into OneDNN logic a bit: it seems like in this case `isa_undef` is equivalent to `avx512_core_amx`: the descriptors are the same except 2 parameters (I checked all tests and models from the latest model cache):

1. `isa_user` expectedly changed from `isa_undef` to `avx512_core_amx`.
2. `sum_dt` unexpectedly differ, but it can be theoretically used only in postops processing which is not used in snippets.

So I think we can set `isa` to `avx512_core_amx` to avoid confusion.
> and it is usually true only when ctx.dt_in0 == dnnl_data_type_t::dnnl_bf16 also true.

That's right. However, we have cases (at least in tests where CPU callback is disabled) in which `ctx.dt_in0 == dnnl_data_type_t::dnnl_bf16` but `ctx.is_with_amx` is false: for instance, when `K` or `N` dims are not divisible by brgemm vnni factor.